packages/headless/src/features/commerce/context/cart/cart-analytics-actions.test.ts

```diff
@@ -0,0 +1,139 @@
+import {createRelay} from '@coveo/relay';
+import {buildCommerceEngine} from '../../../../app/commerce-engine/commerce-engine';
+import {CommerceEngine} from '../../../../app/commerce-engine/commerce-engine';
+import {getSampleCommerceEngineConfiguration} from '../../../../app/commerce-engine/commerce-engine-configuration';
+import {buildMockCommerceState} from '../../../../test/mock-commerce-state';
+import {clearMicrotaskQueue} from '../../../../test/unit-test-utils';
+import {getContextInitialState} from '../context-state';
+import {logCartAction} from './cart-analytics-actions';
+
+jest.mock('@coveo/relay');
+
+describe('Cart Analytics Actions', () => {
+  const emit = jest.fn();
+  let engine: CommerceEngine;
+  const defaultPreloadedState = {
+    ...buildMockCommerceState(),
+    commerceContext: {
+      ...getContextInitialState(),
+      currency: 'USD',
+    },
+  };
+
+  function initEngine(preloadedState = defaultPreloadedState) {
+    engine = buildCommerceEngine({
```

Comment:
Do you have an example of tests for actions that are using mocked engine? (it seems like the dispatch function is a mocked, so It would seems like I cannot test `logCartAction` function).